ts/src/base/Exchange.ts

```diff
@@ -5119,6 +5119,9 @@ export default class Exchange {
                 if (cursorValue === undefined) {
                     break;
                 }
+                if (last['timestamp'] !== undefined && last['timestamp'] < since) {
```

Comment:
we should avoid direct accesses just in case `timestamp` does not exist, it's better to use safeInteger here